src/runtime.js

```diff
@@ -154,7 +163,14 @@ export function adopt(global) {
     registerExtendedTemplate(global, name, implementationClass);
   };
 
-  installRuntimeServices(global);
+  if (!isSingleDoc) {
+    /**
+     * Registers a shadow root document.
+     * @param {!ShadowRoot} shadowRoot
+     */
+    global.AMP.attachShadowRoot = prepareAndAttachShadowRoot.bind(null, global);
```

Comment:
Ah, could this be exposed by `amp-shadow.js` instead. Then it could be dead code eliminated from the other binary.